src/ProgressWorkflow.jsx

```diff
@@ -88,12 +94,11 @@ const ProgressWorkflow = (props) => {
     };
 
     setIsToolbarOpen(!!hasToolbar);
-    const contentId = content?.['@id'];
 
     // filter out paths that don't have workflow (home, login, dexterity even if the content obj stays the same etc)
     if (
       contentId &&
-      contentId.indexOf(basePathname) >= 0 &&
+      contentContainsPathname &&
       basePathname !== '/' && // wihout this there will be a flicker for going back to home ('/' is included in all api paths)
```

Comment:
For the case when we are inside a page but select to go to Site Setup and click back, Volto goes to Home and still shows the progress from the page we were on.
In this care it should be basePathname === "" or pathname === "/"